src/pkg/packager/publish.go

```diff
@@ -0,0 +1,351 @@
+	switch filepath.Ext(filename) {
+	case ".zst":
+		return ZarfLayerMediaTypeTarZstd
+	case ".gz":
+		return ZarfLayerMediaTypeTarGzip
+	case ".yaml":
+		return ZarfLayerMediaTypeYaml
+	case ".json":
+		return ZarfLayerMediaTypeJSON
+	case ".txt":
+		return ZarfLayerMediaTypeTxt
+	default:
+		return ZarfLayerMediaTypeUnknown
+	}
+}
+
+// Publish publishes the package to a registry
+//
+// This is a wrapper around the oras library
+// and much of the code was adapted from the oras CLI - https://github.com/oras-project/oras/blob/main/cmd/oras/push.go
+//
+// Authentication is handled via the Docker config file created w/ `zarf tools registry login`
+func (p *Packager) Publish() error {
+	p.cfg.DeployOpts.PackagePath = p.cfg.PublishOpts.PackagePath
+	if err := p.loadZarfPkg(); err != nil {
+		return fmt.Errorf("unable to load the package: %w", err)
+	}
+
+	paths := []string{
+		p.tmp.ZarfYaml,
+		p.tmp.SbomTar,
+		filepath.Join(p.tmp.Images, "index.json"),
+		filepath.Join(p.tmp.Images, "oci-layout"),
+	}
+	// if checksums.txt file exists, include it
+	if !utils.InvalidPath(filepath.Join(p.tmp.Base, "checksums.txt")) {
+		paths = append(paths, filepath.Join(p.tmp.Base, "checksums.txt"))
+	}
+
+	if p.cfg.Pkg.Kind == "ZarfInitConfig" {
+		seedImagePaths := []string{
+			filepath.Join(p.tmp.SeedImage, "index.json"),
+			filepath.Join(p.tmp.SeedImage, "oci-layout"),
+		}
+		seedImageLayers, err := filepath.Glob(filepath.Join(p.tmp.SeedImage, "blobs", "sha256", "*"))
+		if err != nil {
+			return err
+		}
+		seedImagePaths = append(seedImagePaths, seedImageLayers...)
+		paths = append(paths, seedImagePaths...)
+	}
+	componentDirs, err := filepath.Glob(filepath.Join(p.tmp.Components, "*"))
+	if err != nil {
+		return err
+	}
+	componentTarballs := []string{}
+
+	// repackage the component directories into tarballs
+	for _, componentDir := range componentDirs {
+		dst := filepath.Join(p.tmp.Components, filepath.Base(componentDir)+".tar")
+		err = archiver.Archive([]string{componentDir}, dst)
+		if err != nil {
+			return err
+		}
+		componentTarballs = append(componentTarballs, dst)
+		_ = os.RemoveAll(componentDir)
+	}
+	paths = append(paths, componentTarballs...)
+	imagesLayers, err := filepath.Glob(filepath.Join(p.tmp.Images, "blobs", "sha256", "*"))
+	if err != nil {
+		return err
+	}
+	paths = append(paths, imagesLayers...)
+	ref, err := p.ref("")
+	if err != nil {
+		return err
+	}
+	message.HeaderInfof("📦 PACKAGE PUBLISH %s:%s", p.cfg.Pkg.Metadata.Name, ref.Reference)
+	err = p.publish(ref, paths)
+	if err != nil {
+		return fmt.Errorf("unable to publish package %s: %w", ref, err)
+	}
+	return nil
+}
+
+func (p *Packager) publish(ref registry.Reference, paths []string) error {
+	message.Infof("Publishing package to %s", ref)
+	spinner := message.NewProgressSpinner("")
+	defer spinner.Stop()
+
+	dst, err := utils.NewOrasRemote(ref)
+	if err != nil {
+		return err
+	}
+	ctx := dst.Context
+
+	store, err := file.New(p.tmp.Base)
```

Comment:
store (isn't  very descriptive.  We should probably pick a name that is consistent between it and dst  (even if it is source and destination)